src/main/c/CMakeLists.txt

```
@@ -67,6 +67,14 @@ add_library(seasocks ${SEASOCKS_LIBTYPE}
         util/RootPageHandler.cpp
         $<TARGET_OBJECTS:embedded>
         )
+
+SET_TARGET_PROPERTIES(
+    seasocks
+    PROPERTIES
+        VERSION ${CMAKE_PROJECT_VERSION}
+        SOVERSION ${CMAKE_PROJECT_VERSION}
```

It's better to use the version of the project instead of the global version in case seasocks is vendored and included as a subproject.
```suggestion
        VERSION ${Seasocks_VERSION}}
        SOVERSION ${Seasocks_VERSION}
```

See https://cmake.org/cmake/help/latest/variable/PROJECT-NAME_VERSION.html.
Does seasocks maintain abi compatibility between major versions?
In that case, you can do:
```suggestion
        VERSION ${Seasocks_VERSION}}
        SOVERSION ${Seasocks_MAJOR_VERSION}
```

---

>  Does seasocks maintain abi compatibility between major versions?

In doubt, I wouldn't count on that.